src/ol/interaction/modifyinteraction.js

```diff
@@ -959,6 +1044,12 @@ ol.interaction.Modify.prototype.insertVertex_ = function(segmentData, vertex) {
       coordinates = geometry.getCoordinates();
       coordinates[depth[0]].splice(index + 1, 0, vertex);
       break;
+    case ol.geom.GeometryType.RECTANGLE:
```

Comment:
Ah, right, because I can just push the original segment into this.dragSegments_ and it should come through on drag.